protocol/src/main/java/net/md_5/bungee/protocol/packet/ResourcePackRequest.java

```diff
@@ -0,0 +1,65 @@
+package net.md_5.bungee.protocol.packet;
+
+import io.netty.buffer.ByteBuf;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.chat.ComponentSerializer;
+import net.md_5.bungee.protocol.AbstractPacketHandler;
+import net.md_5.bungee.protocol.DefinedPacket;
+import net.md_5.bungee.protocol.ProtocolConstants;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class ResourcePackRequest extends DefinedPacket
+{
+
+    private String url;
+    private String hash;
+    private boolean forced;
+    private BaseComponent[] promptMessage;
+
+    @Override
+    public void read(ByteBuf buf, ProtocolConstants.Direction direction, int protocolVersion)
+    {
+        boolean applyMaxLength = protocolVersion >= ProtocolConstants.MINECRAFT_1_11;
+        url = applyMaxLength ? readString( buf, 32767 ) : readString( buf );
```

Comment:
32767 is the same as readString(buf) anyway.

What changed in that protocol version?

Reply:
in 1.11+ the protocol mentions max length whilst in <= 1.10 it does not. i'll remove the check for url